src/datasources/alarms/query-type-handlers/alarms-trend/AlarmsTrendQueryHandler.ts

```diff
@@ -1,17 +1,115 @@
 import { DataFrameDTO, DataQueryRequest, FieldType } from '@grafana/data';
 import { AlarmsQueryHandlerCore } from 'datasources/alarms/query-type-handlers/AlarmsQueryHandlerCore';
 import { defaultAlarmsTrendQuery } from 'datasources/alarms/constants/DefaultQueries.constants';
-import { AlarmsTrendQuery } from 'datasources/alarms/types/AlarmsTrend.types';
+import { AlarmsTrendQuery, AlarmTransitionEvent, AlarmTransitionWithNumericTime } from 'datasources/alarms/types/AlarmsTrend.types';
+import { Alarm, AlarmTransitionType, TransitionInclusionOption } from 'datasources/alarms/types/types';
 
 export class AlarmsTrendQueryHandler extends AlarmsQueryHandlerCore {
   public readonly defaultQuery = defaultAlarmsTrendQuery;
 
   public async runQuery(query: AlarmsTrendQuery, options: DataQueryRequest): Promise<DataFrameDTO> {
-    // TODO: alarms trend query logic will be implemented here in future
+    const { start, end, intervalMs } = this.extractTimeParameters(options);
+    const filter = this.getTrendQueryFilter(query, start, end);
+    const alarms = await this.queryAlarmsUntilComplete({ filter, transitionInclusionOption: TransitionInclusionOption.All });
+    const alarmsWithTimestamp = this.enrichTransitionsWithTimestamp(alarms);
+    const trendData = this.countActiveAlarmsPerInterval(alarmsWithTimestamp, start, end, intervalMs);
+
     return {
       refId: query.refId,
-      name: query.refId,
-      fields: [{ name: query.refId, type: FieldType.number, values: [] }],
+      name: 'Alarms Trend',
+      fields: [
+        {
+          name: 'Time',
+          type: FieldType.time,
+          values: Array.from(trendData.keys())
+        },
+        {
+          name: 'Alarms Count',
```

Comment:
Let's follow title case.


```suggestion
          name: 'Alarms count',
```

or 


```suggestion
          name: 'Count',
```